File: app/src/infrastructure/native/bedrock.ts
// AWS Bedrock adapter: delegate a SigV4-authenticated InvokeModel request to
// Rust (credential chain + refresh command live backend-side). Desktop only.
import { invoke } from '@tauri-apps/api/core'
import { isTauri } from './base'

/** Delegate a SigV4-authenticated Bedrock InvokeModel request to Rust. */
export async function bedrockInvoke(region: string, profile: string, refreshCmd: string, credCmd: string, model: string, body: string): Promise<string> {
  if (!isTauri) throw new Error('Bedrock requires the desktop app')
  return await invoke<string>('bedrock_invoke', { region, profile, refreshCmd, credCmd, model, body })
}

File: app/src-tauri/src/commands/bedrock.rs
//! IPC handler for the AWS Bedrock bridge.
use crate::core::bedrock::{self, BedrockState};
use tauri::State;

#[tauri::command]
#[allow(clippy::too_many_arguments)]
pub async fn bedrock_invoke(
    state: State<'_, BedrockState>,
    region: String,
    profile: String,
    refresh_cmd: String,
    cred_cmd: String,
    model: String,
    body: String,
) -> Result<String, String> {
    bedrock::invoke_model(&state, region, profile, refresh_cmd, cred_cmd, model, body).await
}
